src/client/clientenvironment.cpp

```diff
@@ -513,3 +514,8 @@ void ClientEnvironment::getSelectedActiveObjects(
 		}
 	}
 }
+
+void ClientEnvironment::updateFrameTime()
+{
+	m_frame_time = porting::getTimeMs();
```

Comment:
This would be the best location to remind contributors of the purpose of this function.
```suggestion
	// Cache the time each frame to reduce time-consuming getTimeMs() calls for (semi-)debug builds.
	m_frame_time = porting::getTimeMs();
```

Reply:
It's also more correct to always use the same timestamp during rendering of a specific frame.

Reply:
It's an added bonus :)